File: src/Shared/Shared.Contracts/Events/OrderPlacedEvent.cs
namespace Shared.Contracts.Events;

/// <summary>
/// Integration event published when a new order is placed.
/// Consumed by Notification service to trigger confirmation emails.
/// </summary>
public record OrderPlacedEvent(
    Guid OrderId,
    Guid CustomerId,
    decimal TotalAmount,
    DateTime PlacedAt
);

File: src/shared/shared-contracts/src/main/java/com/quickapp/shared/contracts/events/OrderPlacedEvent.java
package com.quickapp.shared.contracts.events;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

/**
 * Integration event published when a new order is placed.
 * Consumed by Notification service to trigger confirmation emails.
 */
public record OrderPlacedEvent(
        UUID orderId,
        UUID customerId,
        BigDecimal totalAmount,
        Instant placedAt) {
}
